### ISSUE TITLE
Issue with StringSliceVar and commas

### ISSUE THREAD

**Issue by adrienkohlbecker** [spf13/pflag#57]
Hello,

We're using `StringSlice` in our CLI and we're having an issue with commas in the values. Here is an example:

```go
// command
tool --env="FOO=BAR" --env="BAZ=a string, with, commas"

// configuration
cmd.cobra.Flags().StringSliceVar(&cmd.additionalEnv, "env", []string{}, "additional environment variables, in the form VARIABLE=VALUE")

// current result
[]string{"FOO=BAR", "BAZ=a string", " with", " commas"}

// expected result
[]string{"FOO=BAR", "BAZ=a string, with, commas"}
```

I noticed the value is split with commas here: https://github.com/spf13/pflag/blob/master/string_slice.go#L24
Is there a way to get around it ? 
We're not using the form `--env=foo,bar,baz`

**Comment by eparis** [comment]
no, it's not possible. I think you'd have to write your own flag type.  I'd also be happy to take a patch which doesn't split on `,` inside `'` or `"`...

**Comment by eparis** [comment]
fixed by #59

**Comment by eparis** [comment]
Remember bash is likely to make things tricky as it will try to strip your `"`, but on the command line `--env=\"foo,bar,baz\",hello,"my,friend"` is going to result in `[]string{"foo,bar,baz", "hello", "my", "friend"}
